src/Aspire.Hosting.Azure/AzureDeployingContext.cs

```diff
@@ -335,6 +344,20 @@ private async Task AuthenticateToAcr(IPublishingStep parentStep, string registry
         }
     }
 
+    private string? GetContainerRuntime()
+    {
+        // Check DcpPublisher configuration first (same as DcpOptions configuration logic)
+        var dcpPublisherConfiguration = configuration.GetSection("DcpPublisher");
+        var dcpConfigValue = dcpPublisherConfiguration["ContainerRuntime"];
+        if (!string.IsNullOrEmpty(dcpConfigValue))
+        {
+            return dcpConfigValue;
+        }
+
```

Comment:
We shouldn't do this in publish mode. DCP is for runtime only.

Reply:
The ship has sailed on this a bit since we've been using DcpOptions to resolve the ContainerRuntime to use in the image builder ([ref](https://github.com/dotnet/aspire/blob/dee0835d4e481dabcce944e7b44fc315172792ee/src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs#L138)). We can change it here but it would be different from the way we resolve the container runtime in other places.

Reply:
Yeah, there's just a chance now that the runtime you use to build might be different from what is configured in the environment variables.